src/test/spec/mod.rs

```diff
@@ -8,16 +8,18 @@ mod read_write_concern;
 use std::{
     ffi::OsStr,
     fs::{self, File},
+    future::Future,
     path::PathBuf,
 };
 
 use bson::Bson;
 use serde::Deserialize;
 use serde_json::Value;
 
-pub(crate) fn run_spec_test<'a, T, F>(spec: &[&str], run_test_file: F)
+pub(crate) async fn run_spec_test<'a, T, F, G>(spec: &[&str], run_test_file: F)
 where
-    F: Fn(T),
+    F: Fn(T) -> G,
```

Comment:
Out of curiosity, would it let you do `F: Fn(T) -> impl Future<Output = ()>`? I'm not sure how impl trait plays in generic bounds, but if ti works, it will save a generic parameter in a function that's started to get a little unwieldy.

Reply:
That's what I originally tried, but unfortunately it fails with the following message:
```
error[E0562]: `impl Trait` not allowed outside of function and inherent method return types
  --> src/test/spec/mod.rs:21:17
   |
21 |     F: Fn(T) -> impl Future<Output = ()>,
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^
```

Reply:
Oh well. Hopefully they eventually make it so this works.